src/network/networkd-routing-policy-rule.c

```diff
@@ -1573,6 +1599,13 @@ static int routing_policy_rule_section_verify(RoutingPolicyRule *rule) {
                 /* rule->family can be AF_UNSPEC only when Family=both. */
         }
 
+        if ((rule->family == AF_INET6 || FLAGS_SET(rule->address_family, ADDRESS_FAMILY_IPV6)) &&
+            (IN_SET(rule->from_type, DHCP4_ADDRESS_TYPE_DHCP4_IP, DHCP4_ADDRESS_TYPE_DHCP4_GATEWAY)))
```

Comment:
`rule->to_type` check is missing.

Reply:
It seems this condition is never gets hit.

```
From=_dhcp4_ip
To=2001:0db8:0100:f101:0210:a4ff:fee3:9566
```

```
To=2001:0db8:0100:f101:0210:a4ff:fee3:9566
From=_dhcp4_ip

```

```
/usr/lib/systemd/network/veth99.network:16: RPDB rule prefix is invalid, ignoring assignment: 2001:0db8:0100:f101:0210:a4ff:fee3:9566
```

The family is already set